ewsonprem_connector.py

```diff
@@ -161,82 +166,61 @@ def _handle_preprocess_scipts(self):
+                "<wsse:messageID>urn:uuid:7f45785a-9691-451e-b3ff-30ab463af64c</wsse:messageID>" \
+                "<wsse:ReplyTo><wsse:Address>http://www.w3.org/2005/08/addressing/anonymous</wsse:Address></wsse:ReplyTo>" \
+                "<wsse:To s:mustUnderstand='1'>{}</wsse:To>" \
+                "<o:Security s:mustUnderstand='1' xmlns:o='http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd'>" \
+                "<u:Timestamp><u:Created>".format(config[EWS_JSON_FED_PING_URL].split('?')[0])
 
             dt_now = datetime.utcnow()
             dt_plus = dt_now + timedelta(minutes=10)
 
             dt_now_str = "{0}Z".format(dt_now.strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3])
             dt_plus_str = "{0}Z".format(dt_plus.strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3])
 
-            ret_val += dt_now_str
-            ret_val += "</u:Created>"
-            ret_val += "<u:Expires>"
-            ret_val += dt_plus_str
-            ret_val += "</u:Expires>"
-            ret_val += "</u:Timestamp>"
-            ret_val += "<o:UsernameToken>"
-            ret_val += "<wsse:Username>"
-            ret_val += config[phantom.APP_JSON_USERNAME]
-            ret_val += "</wsse:Username>"
-            ret_val += "<o:Password>"
-            ret_val += config[phantom.APP_JSON_PASSWORD]
-            ret_val += "</o:Password>"
-            ret_val += "</o:UsernameToken>"
-            ret_val += "</o:Security>"
-            ret_val += "</s:Header>"
-            ret_val += "<s:Body>"
-            ret_val += "<trust:RequestSecurityToken "
-            ret_val += "xmlns:trust='http://docs.oasis-open.org/ws-sx/ws-trust/200512'>"
-            ret_val += "<wsp:AppliesTo xmlns:wsp='http://schemas.xmlsoap.org/ws/2004/09/policy'>"
-            ret_val += "<wsse:EndpointReference><wsse:Address>"
-            ret_val += "urn:federation:MicrosoftOnline</wsse:Address></wsse:EndpointReference>"
-            ret_val += "</wsp:AppliesTo>"
-            ret_val += "<trust:KeyType>http://docs.oasis-open.org/ws-sx/ws-trust/200512/Bearer</trust:KeyType>"
-            ret_val += "<trust:RequestType>http://docs.oasis-open.org/ws-sx/ws-trust/200512/Issue</trust:RequestType>"
-            ret_val += "</trust:RequestSecurityToken>"
-            ret_val += "</s:Body>"
-            ret_val += "</s:Envelope>"
+            ret_val = "{}{}</u:Created>".format(ret_val, dt_now_str)
+            ret_val = "{}<u:Expires>{}</u:Expires></u:Timestamp>".format(ret_val, dt_plus_str)
+
+            ret_val = "{}<o:UsernameToken>".format(ret_val)
+            ret_val = "{}<wsse:Username>{}</wsse:Username>".format(ret_val, config[phantom.APP_JSON_USERNAME])
+            ret_val = "{}<o:Password>{}</o:Password>".format(ret_val, config[phantom.APP_JSON_PASSWORD])
+            ret_val = "{}</o:UsernameToken></o:Security></s:Header>".format(ret_val)
+            ret_val = "{}<s:Body><trust:RequestSecurityToken xmlns:trust='http://docs.oasis-open.org/ws-sx/ws-trust/200512'>" \
```

Comment:
this is kind of hard to read given we keep "overwriting" the same variable but prepending the previous value

since none of these values are conditional, can we just make the whole string (including the large initial definition of ret_val above) into a single constant and then used named formatting to insert the values?

in other words something like:

(in the consts file)
```
SOME_STRING_CONSTANT = "...{ping_url}...{created_date}...{expires_date}...{username}...{password}..."
```

(in this file)
```
SOME_STRING_CONSTANT.format(ping_url=config[EWS_JSON_FED_PING_URL].split('?')[0],
                            created_date=dt_now_str,
                            expires_date=dt_plus_str,
                            username=config[phantom.APP_JSON_USERNAME],
                            password=config[phantom.APP_JSON_PASSWORD])
```

Reply:
Moved the variable to the Constant file and Made an XML string readable.